src/components/implementation/no_interface/cFE_booter/ostask.c

```diff
@@ -0,0 +1,815 @@
+	sl_thd_param_set(main_delegate_thread, sp.v);
+	main_delegate_thread_id = sl_thd_thdid(main_delegate_thread);
+
+	struct cfe_task_info *task_info = &cfe_tasks[main_delegate_thread_id];
+	strcpy(task_info->osal_task_prop.name, "MAIN_THREAD");
+	task_info->osal_task_prop.priority  = MAIN_DELEGATE_THREAD_PRIORITY;
+	task_info->osal_task_prop.OStask_id = (uint32)sl_thd_thdid(main_delegate_thread);
+
+	struct sl_thd *         timer_thd = sl_thd_alloc(timer_fn_1hz, NULL);
+	union sched_param_union spperiod  = {.c = {.type = SCHEDP_WINDOW, .value = HZ_PAUSE}};
+	union sched_param_union spprio    = {.c = {.type = SCHEDP_PRIO, .value = MAIN_DELEGATE_THREAD_PRIORITY + 1}};
+	sl_thd_param_set(timer_thd, spperiod.v);
+	sl_thd_param_set(timer_thd, spprio.v);
+
+	sl_sched_loop();
+}
+
+void
+osal_task_entry_wrapper(void *task_entry)
+{
+	((osal_task_entry)task_entry)();
+}
+
+int
+is_valid_name(const char *name)
+{
+	int i;
+	for (i = 0; i < OS_MAX_API_NAME; i++) {
+		if (name[i] == '\0') { return TRUE; }
+	}
+	return FALSE;
+}
+
+/* TODO: Figure out how to check if thread names are taken */
+int
+is_thread_name_taken(const char *name)
+{
+	return FALSE;
+}
+
+/*
+** Task API
+*/
+
+/* NOTE: Flags here aren't implemented, but I can't find an implementation that does */
+int32
+OS_TaskCreate(uint32 *task_id, const char *task_name, osal_task_entry function_pointer, uint32 *stack_pointer,
+              uint32 stack_size, uint32 priority, uint32 flags)
+{
+	/* TODO: Verify that we don't need to take the cs here */
+
+	if (task_id == NULL || task_name == NULL || function_pointer == NULL) { return OS_INVALID_POINTER; }
+
+	if (!is_valid_name(task_name)) { return OS_ERR_NAME_TOO_LONG; }
+
+	if (is_thread_name_taken(task_name)) { return OS_ERR_NAME_TAKEN; }
+
+	if (priority > 255 || priority < 1) { return OS_ERR_INVALID_PRIORITY; }
+
+	struct sl_thd *thd = sl_thd_alloc(osal_task_entry_wrapper, function_pointer);
+	assert(thd);
+	union sched_param_union sp = {.c = {.type = SCHEDP_PRIO, .value = priority}};
+	sl_thd_param_set(thd, sp.v);
+
+	struct cfe_task_info *task_info = &cfe_tasks[sl_thd_thdid(thd)];
+	strcpy(task_info->osal_task_prop.name, task_name);
+	task_info->osal_task_prop.creator    = OS_TaskGetId();
+	task_info->osal_task_prop.stack_size = stack_size;
+	task_info->osal_task_prop.priority   = priority;
+	task_info->osal_task_prop.OStask_id  = (uint32)sl_thd_thdid(thd);
+	task_info->delete_handler            = NULL;
+
+	*task_id = (uint32)sl_thd_thdid(thd);
+
+	return OS_SUCCESS;
+}
+
+int32
+OS_TaskDelete(uint32 task_id)
+{
+	struct sl_thd *thd = sl_thd_lkup(task_id);
+	if (!thd) { return OS_ERR_INVALID_ID; }
+
+	struct cfe_task_info *task_info = &cfe_tasks[task_id];
+
+	osal_task_entry delete_handler = task_info->delete_handler;
+	if (delete_handler) { delete_handler(); }
+
+	sl_thd_free(thd);
+
+	return OS_SUCCESS;
+}
+
+uint32
+OS_TaskGetId(void)
+{
+	thdid_t real_id           = sl_thdid();
+	thdid_t possible_override = id_overrides[real_id];
```

Comment:
interesting.  not sure where this is set (will grep through), but I'm hoping there is a comment.

Reply:
OK, I understand what it is doing, but it needs a comment when it is set in `launch_other_component`.